stock_picking_package_preparation/models/stock_picking_package_preparation.py

```diff
@@ -0,0 +1,216 @@
+        return company_model._company_default_get(self._name)
+
+    name = fields.Char(
+        related='package_id.name',
+        readonly=True,
+        index=True,
+        store=True,
+    )
+    state = fields.Selection(
+        selection=[('draft', 'Draft'),
+                   ('cancel', 'Cancelled'),
+                   ('in_pack', 'In Pack'),
+                   ('done', 'Done'),
+                   ],
+        default='draft',
+        string='State',
+        readonly=True,
+        copy=False,
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Partner',
+        required=True,
+        states=FIELDS_STATES,
+    )
+    picking_ids = fields.Many2many(
+        comodel_name='stock.picking',
+        relation='stock_picking_pack_prepare_rel',
+        column1='stock_picking_package_preparation_id',
+        column2='stock_picking_id',
+        string='Transfers',
+        copy=False,
+        states=FIELDS_STATES,
+    )
+    packaging_id = fields.Many2one(
+        comodel_name='product.packaging',
+        string='Packaging',
+        states=FIELDS_STATES,
+    )
+    date = fields.Datetime(
+        string='Document Date',
+        default=fields.Datetime.now,
+        states=FIELDS_STATES,
+        copy=False
+    )
+    date_done = fields.Datetime(
+        string='Shipping Date',
+        readonly=True,
+        copy=False,
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
+        required=True,
+        index=True,
+        states=FIELDS_STATES,
+        default=_default_company_id,
+    )
+    move_line_ids = fields.One2many(
+        comodel_name='stock.move.line',
+        compute='_compute_move_line_ids',
+        readonly=True,
+        oldname='pack_operation_ids'
+    )
+    package_id = fields.Many2one(
+        comodel_name='stock.quant.package',
+        string='Pack',
+        readonly=True,
+        copy=False,
+    )
+    note = fields.Text()
+    weight = fields.Float(compute='_compute_weight',
+                          help="The weight is computed when the "
+                               "preparation is done.")
+    quant_ids = fields.Many2many(
+        compute='_compute_quant_ids',
+        comodel_name='stock.quant',
+        relation='stock_quant_pack_prepare_rel',
+        column1='stock_picking_package_preparation_id',
+        column2='stock_quant_id',
+        string='All Content',
+    )
+
+    @api.multi
+    @api.depends('package_id')
+    def _compute_quant_ids(self):
+        for preparation in self:
+            package = preparation.package_id
+            preparation.quant_ids = package._get_contained_quants()
+
+    @api.multi
+    @api.depends('package_id',
+                 'package_id.quant_ids')
+    def _compute_weight(self):
+        for preparation in self:
+            package = preparation.package_id
+            if not package:
+                return
```

Comment:
@SimoRubi I think this should be 'continue' instead as self can be a multi recordset

Reply:
Thanks @rousseldenis, I missed that. Done in https://github.com/OCA/stock-logistics-workflow/pull/448/commits/20954520e6e2decb1728cada9ea7c7b14f583ccb